### PR TITLE
JS: consider quoted string concatenations as sanitizers for js/shell-command-injection-from-environment

### PR DESCRIPTION
**Pull request by erik-krogh**
I encountered [this result](https://lgtm.com/projects/g/jamovi/jamovi-compiler/snapshot/e912724a44de6e149f79b774f2848ccfb6843418/files/index.js?sort=name&dir=ASC&mode=heatmap#xb39ee584ae63c83c:1) in the anomalous query results.    

The shell script is constructed like: 
```JavaScript
let cmd = '"' + source + '" --some --more --hardcoded --args'
```

And I don't think we should flag it.  
If the source is completely attacker controlled, then it's trivial to escape the quotes, but most of what `js/shell-command-injection-from-environment` flags is accidental spaces in paths, which the quotes protects against. 

[An evaluation](https://github.com/dsp-testing/erik-krogh-dca/blob/run/quotedShell-default-shellFromEnv/reports/alert-comparison.md) shows unaffected performance, and that a few results we no longer flag.

### CODE REVIEW

File: javascript/ql/src/semmle/javascript/security/dataflow/ShellCommandInjectionFromEnvironmentCustomizations.qll [hunk 1]
```diff
@@ -55,4 +55,14 @@ module ShellCommandInjectionFromEnvironment {
   class ShellCommandSink extends Sink, DataFlow::ValueNode {
     ShellCommandSink() { any(SystemCommandExecution sys).isShellInterpreted(this) }
   }
+
+  /**
+   * A string-concatenation leaf that is sorounded by quotes, seen as a sanitizer for command-injection.
```
Comment (intrigus-lgtm) [inline javascript/ql/src/semmle/javascript/security/dataflow/ShellCommandInjectionFromEnvironmentCustomizations.qll:60]:
```suggestion
   * A string-concatenation leaf that is surrounded by quotes, seen as a sanitizer for command-injection.
```

**Comment by esbena** [comment]
I am inclined to agree.
Do you see any way of recovering any lost TPs by recognizing some of these quote wrapping as bad sanitization attempts? Something like the below example should ideally still be flagged by some query.

```js
let safeSource = source.replace(/`/g, "");
let cmd = '"' + safeSource + '" --some --more --hardcoded --args'
...
```

**Comment by erik-krogh** [comment]
> Do you see any way of recovering any lost TPs by recognizing some of these quote wrapping as bad sanitization attempts? Something like the below example should ideally still be flagged by some query.

None of the lost results from the evaluation appear to have bad sanitizers associated. They don't have any sanitizer at all, and I don't think they need one.   
(I didn't look through the testcases from `nodejs/node`).   

We could try to recognize bad shell sanitizers.   
But I think it should only consider strings that end in a shell-script execution to avoid noise.  
(That is basically the `js/shell-command-constructed-from-input` sinks).

**Comment by erik-krogh** [comment]
> I could check that out....

Yeah. A quick-and-dirty attempt at a bad-sanitizer query: https://lgtm.com/query/8953307798337762059/
Just looking at quote-sanitizers, without checking that it ends in a shell-execution, is too noisy.

**Review by esbena — APPROVED** [review]
Thanks for checking.

**Pull request merged**